external/llvm-project/mlir/lib/Dialect/GPU/Transforms/SerializeToHsaco.cpp

```diff
@@ -347,6 +338,13 @@ SerializeToHsacoPass::translateToLLVMIR(llvm::LLVMContext &llvmContext) {
       return nullptr;
     }
   }
+
+  // Set amdgpu_hostcall if host calls have been linked, as needed by newer LLVM
```

Comment:
Necessary for Krzysztof's gpu.printf.  He says it may not be necessary with the most recent upstream, but I'll leave it for him to fix on the next cycle.